cms/djangoapps/contentstore/rest_api/v1/views.py

```diff
@@ -58,45 +62,106 @@ def proctored_exam_settings(request, course_id):
             "course_start_date": "2013-02-05T05:00:00Z",
             "is_staff": true
         }
+
+    ------------------------------------------------------------------------------------
+    POST
+    ------------------------------------------------------------------------------------
+
+    **Returns**
+
+        * 200: OK - Contains a set of course proctored exam settings.
+        * 400: Bad Request - Unable to save requested settings.
+        * 401: The requesting user is not authenticated.
+        * 403: The requesting user lacks access to the course.
+        * 404: The requested course does not exist.
+
+    **Response**
+
+        In the case of a 200 response code, the response will echo the updated proctored
+        exam settings data.
     """
-    course_key = CourseKey.from_string(course_id)
-    with modulestore().bulk_operations(course_key):
-        if request.method == 'GET':
-            course_module = get_course_and_check_access(course_key, request.user)
+    PROCTORED_EXAM_SETTINGS_KEYS = [
+        'enable_proctored_exams',
+        'allow_proctoring_opt_out',
+        'proctoring_provider',
+        'proctoring_escalation_email',
+        'create_zendesk_tickets',
+    ]
+
+    def get(self, request, course_id):
+        """ GET handler """
+        course_module = self._get_and_validate_course(request.user, course_id)
+        course_metadata = CourseMetadata().fetch_all(course_module)
+        proctored_exam_settings = self._get_proctored_exam_setting_values(course_metadata)
+
+        data = {}
+
+        data['proctored_exam_settings'] = proctored_exam_settings
+        data['available_proctoring_providers'] = get_available_providers()
+        data['course_start_date'] = course_metadata['start'].get('value')
+        data['is_staff'] = request.user.is_staff
+
+        serializer = ProctoredExamConfigurationSerializer(data)
+
+        return Response(serializer.data)
+
+    def post(self, request, course_id):
+        """ POST handler """
+        exam_config = ProctoredExamSettingsSerializer(data=request.data.get('proctored_exam_settings', {}))
+        exam_config.is_valid(raise_exception=True)
+        with modulestore().bulk_operations(CourseKey.from_string(course_id)):
+            course_module = self._get_and_validate_course(request.user, course_id)
+            course_metadata = CourseMetadata().fetch_all(course_module)
 
-            if not course_module:
+            models_to_update = {}
+            for setting_key, value in exam_config.data.items():
```

Comment:
This ignores superfluous data passed in the `POST` request. Is that what we want? I think that if a request is made with extra data in the body, we should return a `HTTP_400_BAD_REQUEST ` response.

Reply:
those extra fields would already be dropped since it's serialized as ProctoredExamSettingsSerializer.  There may be a way to make that more strict but I don't think that is a convention we follow elsewhere.